### ISSUE TITLE
Build inconsistency on Windows since 3.10.8

### ISSUE THREAD

**Issue by Jeroen52** [pawn-lang/compiler#364]
### Issue description: Compile inconsistencies between Linux and Windows since 3.10.8

I'm having compiling inconsistencies on Windows binaries compared to the Linux release,

I'm not really that bothered with stricter coding standards with 3.10.8, but I do get the errors on Linux using my CI system but those errors don't occur on Windows.

The compile errors also happen when I just tar the project directory on Windows where it can compile, transfer it over to a Linux machine and compile it there.

Compile on Windows.
```
Pawn compiler 3.10.8	 	 	Copyright (c) 1997-2006, ITB CompuPhase

Header size:          44176 bytes
Code size:          3769092 bytes
Data size:          6775544 bytes
Stack/heap size:      16384 bytes; estimated max. usage=3190 cells (12760 bytes)
Total requirements:10605196 bytes
```

Compile on Linux
```
../pawno/include/bcrypt-samp\pawn\bcrypt.inc(74) : error 239: literal array/string passed to a non-const parameter
../pawno/include/fixes.inc(4433) : error 239: literal array/string passed to a non-const parameter
../pawno/include/fixes.inc(4463) : error 239: literal array/string passed to a non-const parameter
../pawno/include/timerfix.inc(60) : error 239: literal array/string passed to a non-const parameter
../pawno/include/irc.inc(68) : error 214: possibly a "const" array argument was intended: "user"
../pawno/include/irc.inc(84) : error 214: possibly a "const" array argument was intended: "user"
../pawno/include/irc.inc(101) : error 214: possibly a "const" array argument was intended: "user"
../pawno/include/irc.inc(118) : error 214: possibly a "const" array argument was intended: "user"
../pawno/include/irc.inc(135) : error 214: possibly a "const" array argument was intended: "user"
../pawno/include/sscanf2.inc(305) : error 214: possibly a "const" array argument was intended: "string"
../pawno/include/sscanf2.inc(365) : error 214: possibly a "const" array argument was intended: "string"
../pawno/include/mSelection.inc(103) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(110) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(110) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(119) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(172) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(191) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(221) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(230) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(232) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(233) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(234) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(235) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(256) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(287) : error 239: literal array/string passed to a non-const parameter
../pawno/include/mSelection.inc(292) : error 239: literal array/string passed to a non-const parameter

Compilation aborted.

Pawn compiler 3.10.8	 	 	Copyright (c) 1997-2006, ITB CompuPhase


26 Errors.
```

### Workspace Information:

* Compiler version: 3.10.8
* Command line arguments provided (or sampctl version):  `pawncc "-;+" "-(+" -i:../pawno/include SFCRRPG.pwn -d3 -Z -E`
* Windows command line arguments:
```
	"build_systems":
	[
		{
			"cmd":
			[
				"$project/../pawno/pawncc.exe",
				"-;+",
				"-(+",
				"-i:../pawno/include",
				"$file",
				"-d3",
				"-Z",
				"-E"
			],
			"name": "Pawn",
			"quiet": true
		}
	],
```
* Operating System: CentOS 7 (running via Docker + GitLab Runner), Windows 7.

**Comment by Y-Less** [comment]
That's not an inconsistency, that's just a different version.  Your Linux compiler is using git head instead of 3.10.8 release.

**Comment by Jeroen52** [comment]
I am not using git head. This is my Dockerfile.
```Dockerfile
FROM centos:7

MAINTAINER jeroen@jeroendeneef.com

ENV VERSION 3.10.8
ENV PAWNCFOLDER ~/pawn

LABEL	name="Pawn compiler" \
		vendor="pawn-lang" \
		license="zLib/libpng"

RUN	yum -y install \
			deltarpm epel-release; \
	yum -y groupinstall \
			'Development Tools'; \
	curl -s \
		https://packagecloud.io/install/repositories/github/git-lfs/script.rpm.sh \
		| bash; \
	yum -y install \
			kernel-devel git glibc.i686 \
			libuuid libuuid-devel openssl \
			openssl-devel wget libgcc.i686 \
			gcc gcc-c++ cmake28 \
			cmake make glibc-devel.i686 \
			libgcc.i686 libstdc++-devel.i686 \
			openssh git-lfs ; \
	yum -y update; \
	yum -y clean \
			all; \
	git lfs install;

RUN		git clone https://github.com/pawn-lang/compiler.git $PAWNCFOLDER; \
		cd $PAWNCFOLDER; \
		git checkout tags/v$VERSION; \
		mkdir build && cd build; \
		cmake ../source/compiler -DCMAKE_C_FLAGS=-m32 -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF; \
		make; \
		mv libpawnc.so /lib/; \
		mv pawndisasm /bin/; \
		mv pawncc /bin/; \
		cd ~; \
		rm -rf $PAWNCFOLDER/;


```

As you can see, I'm cloning the repo, checking out the tagged version and then building it from that version.

**Comment by Y-Less** [comment]
No, that's since the 3.10.8 branch was merged back in to master, not since 3.10.8 was forked, which was around here:

https://github.com/pawn-lang/compiler/commit/fd4369e67b2c9bbf5b9fbceb277a1d6f1f61cab4
